### PR TITLE
Revenue tracking: Breakdown queries

### PR DESCRIPTION
**Pull request by vinibrsl**

### CODE REVIEW

File: lib/plausible/stats/base.ex [hunk 1]
```diff
@@ -277,6 +277,42 @@ defmodule Plausible.Stats.Base do
...
+      }
+    )
+    |> select_event_metrics(rest)
+  end
+
+  def select_event_metrics(q, [:average_revenue | rest]) do
+    from(e in q,
+      select_merge: %{
+        average_revenue:
+          type(
```
Comment (vinibrsl) [inline lib/plausible/stats/base.ex:302]: I'm getting this error when calling `type(fragment(...), e.revenue_reporting_amount)`:

```
  3) test GET /api/stats/:domain/conversions returns revenue metrics as nil for non-revenue goals (PlausibleWeb.Api.StatsController.ConversionsTest)
     test/plausible_web/controllers/api/stats_controller/conversions_test.exs:257
     ** (ArgumentError) unknown or ambiguous ClickHouse type: :any
     stacktrace:
       (ecto_ch 0.1.9) lib/ecto/adapters/clickhouse/connection.ex:879: Ecto.Adapters.ClickHouse.Connection.ecto_to_db/1
       (ecto_ch 0.1.9) lib/ecto/adapters/clickhouse/connection.ex:696: Ecto.Adapters.ClickHouse.Connection.expr/4
       (ecto_ch 0.1.9) lib/ecto/adapters/clickhouse/connection.ex:774: Ecto.Adapters.ClickHouse.Connection.intersperse_map/3
       (ecto_ch 0.1.9) lib/ecto/adapters/clickhouse/connection.ex:774: Ecto.Adapters.ClickHouse.Connection.intersperse_map/3
       (ecto_ch 0.1.9) lib/ecto/adapters/clickhouse/connection.ex:243: Ecto.Adapters.ClickHouse.Connection.select/3
       (ecto_ch 0.1.9) lib/ecto/adapters/clickhouse/connection.ex:59: Ecto.Adapters.ClickHouse.Connection.all/3
       (ecto_ch 0.1.9) lib/ecto/adapters/clickhouse.ex:258: Ecto.Adapters.ClickHouse.execute/5
       (ecto 3.9.5) lib/ecto/repo/queryable.ex:229: Ecto.Repo.Queryable.execute/4
       (ecto 3.9.5) lib/ecto/repo/queryable.ex:19: Ecto.Repo.Queryable.all/3
       (plausible 0.0.1) lib/plausible/stats/breakdown.ex:240: Plausible.Stats.Breakdown.breakdown_events/5
       (plausible 0.0.1) lib/plausible/stats/breakdown.ex:36: Plausible.Stats.Breakdown.breakdown/5
       (plausible 0.0.1) lib/plausible_web/controllers/api/stats_controller.ex:1051: PlausibleWeb.Api.StatsController.conversions/2
       (plausible 0.0.1) lib/plausible_web/controllers/api/stats_controller.ex:1: PlausibleWeb.Api.StatsController.action/2
       (plausible 0.0.1) lib/plausible_web/controllers/api/stats_controller.ex:1: PlausibleWeb.Api.StatsController.phoenix_controller_pipeline/2
       (plausible 0.0.1) lib/plug/error_handler.ex:80: PlausibleWeb.Api.StatsController.call/2
       (phoenix 1.7.2) lib/phoenix/router.ex:430: Phoenix.Router.__call__/5
       (plausible 0.0.1) lib/plausible_web/endpoint.ex:1: PlausibleWeb.Endpoint.plug_builder_call/2
       (plausible 0.0.1) lib/plausible_web/endpoint.ex:1: PlausibleWeb.Endpoint."call (overridable 3)"/2
       (plausible 0.0.1) lib/plausible_web/endpoint.ex:1: PlausibleWeb.Endpoint.call/2
       (phoenix 1.7.2) lib/phoenix/test/conn_test.ex:225: Phoenix.ConnTest.dispatch/5
```

This query returns a ClickHouse integer, but I want to load it using my custom type `Plausible.MoneyWithoutCurrency`. I think I got something wrong with `Plausible.MoneyWithoutCurrency.type/1`. It's currently defined as `:u64`, but I'm not sure if that's correct.

https://github.com/plausible/analytics/pull/2992/files#diff-49df18140f1f5a5d42c296336875a3b941c2d8c90cda661840e92e14d58da3a8L27

@ruslandoga could you please give me a hand figuring out what's going on?

Comment (ruslandoga) [reply lib/plausible/stats/base.ex:302]: 👋 @vinibrsl 

It's a bug in `ecto_ch` -- I didn't expect tagged values to come in as `:any`, quickfix: https://github.com/plausible/ecto_ch/pull/78

```diff
def deps do
  [
-   {:ecto_ch, "~> 0.1.0"},
+   {:ecto_ch, github: "plausible/ecto_ch", branch: "allow-any-tagged-values"},
  ]
end
```

Comment (ruslandoga) [reply lib/plausible/stats/base.ex:302]: I'll do a proper fix with tests tomorrow and release a new version.

Comment (vinibrsl) [reply lib/plausible/stats/base.ex:302]: Thank you!

Comment (vinibrsl) [reply lib/plausible/stats/base.ex:302]: I pushed some wip changes. The `type` call is not failing anymore (thank you for the quick fix!). However, values are returned as integers, and I was expecting `Plausible.MoneyWithoutCurrency.load/3` to be called and convert them to `%Decimal{}`s.

```
%{average_revenue: 166733566748666, total_revenue: 500200700246000, ...}
```

As you played with Ecto more than me, is this something with my type definition?

Comment (ruslandoga) [reply lib/plausible/stats/base.ex:302]: I think base_query is "schemaless". If so, then Ecto doesn't really know about your custom type during loading the rows into structs.

Comment (ruslandoga) [reply lib/plausible/stats/base.ex:302]: Maybe something like https://github.com/plausible/analytics/pull/2995 would work.

Comment (vinibrsl) [reply lib/plausible/stats/base.ex:304]: That worked perfectly. Thank you!

Comment (ruslandoga) [reply lib/plausible/stats/base.ex:304]: @vinibrsl I've published the fix in `v0.1.10`

**Pull request merged**